syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.betasolutions.grpc.chat";
option java_outer_classname = "ChatProto";
option objc_class_prefix = "RTG";

package chat;


service Chat {
    rpc realTimeChat (stream UserMessage) returns (stream UserMessage) {
    }

    rpc listUserMessages (ListMessagesRequest) returns (stream UserMessage) {
    }

}
message UserMessage {
    string username = 1;
    string text_message = 2;
    int64 id = 3;
    string topic = 4;
    string client_id = 5;
}

message ListMessagesRequest {
    int64 current_id = 1;
    int32 max_page_size = 2;
    string topic = 3;
    string client_id = 4;
}